Hal doesn't build with unproven features for atsamd21e18a

Compiling current master (and it seems also the releases) for the samd21e18a target with the unproven feature errors out
```
atsamd/hal [master] λ cargo build --features unproven,samd21e18a
[...]
   Compiling atsamd-hal v0.7.4 (/home/soryio/usr/proj/181127-embedded_rust-69/atsamd/hal)
error[E0432]: unresolved imports `crate::gpio::Pb0`, `crate::gpio::Pb1`, `crate::gpio::Pb2`, `crate::gpio::Pb3`, `crate::gpio::Pb4`, `crate::gpio::Pb5`, `crate::gpio::Pb6`, `crate::gpio::Pb7`, `crate::gpio::Pb8`, `crate::gpio::Pb9`
 --> src/samd21/adc.rs:3:57
  |
3 |     Pa10, Pa11, Pa2, Pa3, Pa4, Pa5, Pa6, Pa7, Pa8, Pa9, Pb0, Pb1, Pb2, Pb3, Pb4, Pb5, Pb6, Pb7,
  |                                                         ^^^  ^^^  ^^^  ^^^  ^^^  ^^^  ^^^  ^^^ no `Pb7` in `common::gpio`
  |                                                         |    |    |    |    |    |    |
  |                                                         |    |    |    |    |    |    no `Pb6` in `common::gpio`
  |                                                         |    |    |    |    |    no `Pb5` in `common::gpio`
  |                                                         |    |    |    |    no `Pb4` in `common::gpio`
  |                                                         |    |    |    no `Pb3` in `common::gpio`
  |                                                         |    |    no `Pb2` in `common::gpio`
  |                                                         |    no `Pb1` in `common::gpio`
  |                                                         no `Pb0` in `common::gpio`
4 |     Pb8, Pb9, PfB,
  |     ^^^  ^^^ no `Pb9` in `common::gpio`
  |     |
  |     no `Pb8` in `common::gpio`
help: a similar name exists in the module
  |
3 |     Pa10, Pa11, Pa2, Pa3, Pa4, Pa5, Pa6, Pa7, Pa8, Pa9, Pa0, Pb1, Pb2, Pb3, Pb4, Pb5, Pb6, Pb7,
  |                                                         ^^^
help: a similar name exists in the module
  |
3 |     Pa10, Pa11, Pa2, Pa3, Pa4, Pa5, Pa6, Pa7, Pa8, Pa9, Pb0, Pa1, Pb2, Pb3, Pb4, Pb5, Pb6, Pb7,
  |                                                              ^^^
help: a similar name exists in the module
  |
3 |     Pa10, Pa11, Pa2, Pa3, Pa4, Pa5, Pa6, Pa7, Pa8, Pa9, Pb0, Pb1, Pa2, Pb3, Pb4, Pb5, Pb6, Pb7,
  |                                                                   ^^^
help: a similar name exists in the module
  |
3 |     Pa10, Pa11, Pa2, Pa3, Pa4, Pa5, Pa6, Pa7, Pa8, Pa9, Pb0, Pb1, Pb2, Pa3, Pb4, Pb5, Pb6, Pb7,
  |                                                                        ^^^
help: a similar name exists in the module
  |
3 |     Pa10, Pa11, Pa2, Pa3, Pa4, Pa5, Pa6, Pa7, Pa8, Pa9, Pb0, Pb1, Pb2, Pb3, Pa4, Pb5, Pb6, Pb7,
  |                                                                             ^^^
help: a similar name exists in the module
  |
3 |     Pa10, Pa11, Pa2, Pa3, Pa4, Pa5, Pa6, Pa7, Pa8, Pa9, Pb0, Pb1, Pb2, Pb3, Pb4, Pa5, Pb6, Pb7,
  |                                                                                  ^^^
help: a similar name exists in the module
  |
3 |     Pa10, Pa11, Pa2, Pa3, Pa4, Pa5, Pa6, Pa7, Pa8, Pa9, Pb0, Pb1, Pb2, Pb3, Pb4, Pb5, Pa6, Pb7,
  |                                                                                       ^^^
help: a similar name exists in the module
  |
3 |     Pa10, Pa11, Pa2, Pa3, Pa4, Pa5, Pa6, Pa7, Pa8, Pa9, Pb0, Pb1, Pb2, Pb3, Pb4, Pb5, Pb6, Pa7,
  |                                                                                            ^^^
help: a similar name exists in the module
  |
4 |     Pa8, Pb9, PfB,
  |     ^^^
help: a similar name exists in the module
  |
4 |     Pb8, Pa9, PfB,
  |          ^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0432`.
error: could not compile `atsamd-hal`.

To learn more, run the command again with --verbose.
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -14,7 +14,7 @@ env:
   - CRATE=boards/gemma_m0
   - CRATE=boards/itsybitsy_m0
   - CRATE=boards/itsybitsy_m4 FEATURES="--features=unproven,usb,use_uart_debug"
-  - CRATE=boards/trinket_m0
+  - CRATE=boards/trinket_m0 FEATURES="--features=unproven"
   - CRATE=boards/samd11_bare FEATURES="--features=unproven" BUILDMODE="--release"
   - CRATE=boards/samd21_mini
   - CRATE=boards/arduino_mkrzero

--- a/hal/src/samd21/adc.rs
+++ b/hal/src/samd21/adc.rs
@@ -1,8 +1,7 @@
 use crate::clock::GenericClockController;
-use crate::gpio::{
-    Pa10, Pa11, Pa2, Pa3, Pa4, Pa5, Pa6, Pa7, Pa8, Pa9, Pb0, Pb1, Pb2, Pb3, Pb4, Pb5, Pb6, Pb7,
-    Pb8, Pb9, PfB,
-};
+use crate::gpio::{Pa10, Pa11, Pa2, Pa3, Pa4, Pa5, Pa6, Pa7, Pa8, Pa9, PfB};
+#[cfg(any(feature = "samd21g18a", feature = "samd21j18a"))]
+use crate::gpio::{Pb0, Pb1, Pb2, Pb3, Pb4, Pb5, Pb6, Pb7, Pb8, Pb9};
 use crate::hal::adc::{Channel, OneShot};
 use crate::target_device::{adc, ADC, PM};
 
@@ -122,12 +121,18 @@ impl Channel<ADC> for $pin<PfB> {
 adc_pins! {
     Pa2: 0,
     Pa3: 1,
-    Pb8: 2,
-    Pb9: 3,
     Pa4: 4,
     Pa5: 5,
     Pa6: 6,
     Pa7: 7,
+    Pa8: 16,
+    Pa9: 17,
+    Pa10: 18,
+    Pa11: 19
+}
+
+#[cfg(any(feature = "samd21g18a", feature = "samd21j18a"))]
+adc_pins! {
     Pb0: 8,
     Pb1: 9,
     Pb2: 10,
@@ -136,8 +141,6 @@ adc_pins! {
     Pb5: 13,
     Pb6: 14,
     Pb7: 15,
-    Pa8: 16,
-    Pa9: 17,
-    Pa10: 18,
-    Pa11: 19
+    Pb8: 2,
+    Pb9: 3
 }

